aws/resource_aws_mwaa_environment_test.go

```diff
@@ -0,0 +1,770 @@
+					resource.TestCheckResourceAttr(resourceName, "network_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "network_configuration.0.security_group_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "network_configuration.0.subnet_ids.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "plugins_s3_object_version", ""),
+					resource.TestCheckResourceAttr(resourceName, "plugins_s3_path", "plugins.zip"),
+					resource.TestCheckResourceAttr(resourceName, "requirements_s3_object_version", ""),
+					resource.TestCheckResourceAttr(resourceName, "requirements_s3_path", "requirements.txt"),
+					resource.TestCheckResourceAttrSet(resourceName, "service_role_arn"),
+					testAccCheckResourceAttrGlobalARNNoAccount(resourceName, "source_bucket_arn", "s3", rName),
+					resource.TestCheckResourceAttrSet(resourceName, "status"),
+					resource.TestCheckResourceAttr(resourceName, "webserver_access_mode", mwaa.WebserverAccessModePublicOnly),
+					resource.TestCheckResourceAttrSet(resourceName, "webserver_url"),
+					resource.TestCheckResourceAttr(resourceName, "weekly_maintenance_window_start", "SAT:03:00"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Name", rName),
+					resource.TestCheckResourceAttr(resourceName, "tags.Environment", "production"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSMwaaEnvironmentExists(resourceName string, environment *mwaa.GetEnvironmentOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No MWAA Environment ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).mwaaconn
+		resp, err := conn.GetEnvironment(&mwaa.GetEnvironmentInput{
+			Name: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return fmt.Errorf("Error getting MWAA Environment: %s", err.Error())
+		}
+
+		*environment = *resp
+
+		return nil
+	}
+}
+
+func testAccCheckAWSMwaaEnvironmentDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).mwaaconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_mwaa_environment" {
+			continue
+		}
+
+		_, err := conn.GetEnvironment(&mwaa.GetEnvironmentInput{
+			Name: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			if isAWSErr(err, mwaa.ErrCodeResourceNotFoundException, "") {
+				return nil
+			}
+			return err
+		}
+
+		return fmt.Errorf("Expected MWAA Environment to be destroyed, %s found", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func testAccAWSMwaaEnvironmentBase(rName string) string {
+	return fmt.Sprintf(`
+data "aws_availability_zones" "available" {
+  state = "available"
+
+  filter {
+    name   = "opt-in-status"
+    values = ["opt-in-not-required"]
+  }
+}
+
```

Comment:
Recommend using `aws_partition` data source to prevent hardcoding configuration such as DNS suffixes

```suggestion

data "aws_partition" "current" {}

```